dataplug-twitter/app/org/hatdex/dataplugTwitter/apiInterfaces/TwitterFriendInterface.scala

```diff
@@ -112,31 +150,43 @@ class TwitterFriendInterface @Inject() (
     fetchParameters: ApiEndpointCall)(implicit ec: ExecutionContext, timeout: Timeout): Future[Done] = {
 
     for {
-      users <- FutureTransformations.transform(validateMinDataStructure(content)) // Parse tweets into strongly-typed structures
+      clippedData <- transformData(content, fetchParameters)
+      users <- FutureTransformations.transform(validateMinDataStructure(clippedData)) // Parse tweets into strongly-typed structures
       _ <- uploadHatData(namespace, endpoint, users, hatAddress, hatClient) // Upload the data
     } yield {
       logger.debug(s"Successfully synced new records for HAT $hatAddress")
       Done
     }
   }
 
+  private def transformData(content: JsValue, params: ApiEndpointCall): Future[JsValue] = {
```

Comment:
not a great function name semantically, suggest renaming to `takeNew` or similar